test/recurly/BaseClient.test.js

```diff
@@ -32,4 +34,165 @@ describe('BaseClient', () => {
       assert.equal(path, '/accounts/code-benjamin%20du%20monde/shipping_addresses/1234567890')
     })
   })
+
+  describe('with mocked request adapter', () => {
+    beforeEach(() => {
+      client.mock((resp, options) => {
+        if (options.path === '/resources/myid') {
+          resp.status = 200
+          resp.body = { id: 'myid', object: 'my_resource' }
+        } else {
+          resp.status = 404
+          resp.body = { error: { type: 'not_found' } }
+        }
+        return Promise.resolve(resp)
+      })
+    })
+
+    describe('#getResource', () => {
+      it('Should return a resource given a valid id', () => {
+        const resp = client.getResource('myid')
+        return resp
+          .then(resource => {
+            assert(resource instanceof MyResource)
+
+            let options = sinon.match({
+              method: 'GET',
+              path: '/resources/myid'
+            })
+            assert(client.calledWith(options, null))
```

Comment:
We want all mocked client tests to do 2 things:

1. Verify that the correct arguments were applied to the `requestAdapter`.
2. Verify that given different responses from `requestAdapter`, the our code correctly handled and transformed each case.